synapse/federation/transport/server.py

```diff
@@ -396,6 +407,24 @@ def on_POST(self, origin, content, query, room_id):
         defer.returnValue((200, content))
 
 
+class On3pidBindServlet(BaseFederationServlet):
+    PATH = "/3pid/onbind"
+
+    @defer.inlineCallbacks
+    def on_POST(self, request):
+        content_bytes = request.content.read()
+        content = json.loads(content_bytes)
+        if "invites" in content:
+            for invite in content["invites"]:
+                yield self.handler.exchange_third_party_invite(invite)
+        defer.returnValue((200, {}))
```

Comment:
What would you hand to federation_server? The list of invites? It doesn't seem worthwhile to have this chain of methods:

#Transport server
@defer.inlineCallbacks
def on_POST(self, request):
    content_bytes = request.content.read()
    content = json.loads(content_bytes)
    if "invites" in content:
        yield self.handler.on_3pid_bind(content["invites"])
    defer.returnValue((200, {}))

# Federation_server
def on_3pid_bind(invites):
    for invite in invites:
        yield self.handler.exchange_third_party_invite(invite)

# Federation handler
def exchange_third_party_invite(self, invite):
    # ...